tasks/main.yml

```diff
@@ -15,25 +11,76 @@
     - "{{ ansible_facts['distribution'] ~ '_' ~
           ansible_facts['distribution_version'] }}.yml"
   vars:
-    __template_vars_file: "{{ role_path }}/vars/{{ item }}"
-  when: __template_vars_file is file
+    __crypto_policies_vars_file: "{{ role_path }}/vars/{{ item }}"
+  when: __crypto_policies_vars_file is file
+
+- name: OS is supported
+  meta: end_host
+  when:
+    - not __crypto_policies_os_supported|bool
 
-# Examples of some tasks:
 - name: Ensure required packages are installed
   package:
-    name: "{{ __template_packages }}"
+    name: "{{ __crypto_policies_packages }}"
     state: present
 
-- name: Ensure required services are enabled and started
-  service:
-    name: "{{ item }}"
-    state: started
-    enabled: yes
-  loop: "{{ __template_services }}"
-
-- name: Generate /etc/{{ __template_foo_config }}
-  template:
-    src: "{{ __template_foo_config }}.j2"
-    dest: /etc/{{ __template_foo_config }}
-    backup: yes
-  notify: template handler to restart services
+- include_tasks: gather_facts.yml
+
+- name: Find out what base policies are available
+  shell: >
+    set -euo pipefail && (
+      ls -q /usr/share/crypto-policies/policies/*pol;
+      ls -q /etc/crypto-policies/policies/*pol || true
+    ) 2>/dev/null | xargs -n 1 basename | cut -d '.' -f1 |  sort | uniq
+  changed_when: false
+  register: __crypto_policies_available_policies
+
+- name: Set available base policies fact
+  set_fact:
+    crypto_policies_available_policies: >-
+      {{ __crypto_policies_available_policies.stdout_lines }}
+    cacheable: yes
+
+- name: Find out what policy modules are available
+  shell: >
+    set -euo pipefail && (
+      ls -q /usr/share/crypto-policies/policies/modules/*pmod;
+      ls -q /etc/crypto-policies/policies/modules/*pmod || true
+    ) 2>/dev/null | xargs -n 1 basename | cut -d '.' -f1 |  sort | uniq
+  changed_when: false
+  register: __crypto_policies_available_modules
+
+- name: Set available policy modules fact
+  set_fact:
+    crypto_policies_available_modules: >-
+      {{ __crypto_policies_available_modules.stdout_lines }}
+    cacheable: yes
+
+- debug:
+    var: "crypto_policies_active"
+- debug:
+    var: "crypto_policies_policy"
+- debug:
+    var: "crypto_policies_policy|bool"
+
+- name: Update crypto policy if needed
+  command: >
+    update-crypto-policies
+      {{ '--no-reload' if not crypto_policies_reload }}
+      --set {{ crypto_policies_policy }}
+  when:
+    # TODO check with available policies and modules
+    - crypto_policies_policy != false
```

Comment:
This is something linters complain about (comparison against `false` value) so I assume I am doing something wrong. The idea is to have a default value that will just report facts as active policy and available polcies (see readme) and if this variable is set to something else (policy name), it is applied.
Q: Is it a good idea to do something like this?
Q2: Is there a better way how to handle a varialbes that can be either undefined/false or have a set value?